framework/src/controller/application_state.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable class-methods-use-this */
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const os = require('os');
+const crypto = require('crypto');
+const _ = require('lodash');
+
+/**
+ * Initial state of the entire application.
+ * - os
+ * - version
+ * - wsPort
+ * - httpPort
+ * - minVersion
+ * - protocolVersion
+ * - height
+ * - nethash
+ * - broadhash
+ * - nonce
+ *
+ * @class
+ * @requires crypto
+ * @requires os
+ * @param {Object} initialState - Initial state of the application
+ * @param {Object} logger
+ */
+class ApplicationState {
+	constructor(initialState, logger) {
+		this.logger = logger;
+		this.state = {
+			os: os.platform() + os.release(),
+			version: initialState.version,
+			wsPort: initialState.wsPort,
+			httpPort: initialState.httpPort,
+			minVersion: initialState.minVersion,
+			protocolVersion: initialState.protocolVersion,
+			height: 1,
+			nethash: initialState.nethash,
+			broadhash: initialState.nethash,
+			nonce: initialState.nonce,
+		};
+	}
+
+	getState() {
+		const state = _.cloneDeep(this.state);
+		return state;
+	}
+
+	setChannel(channel) {
+		this.channel = channel;
+	}
+
+	/**
+	 * Updates broadhash and height values.
+	 *
+	 * @returns {Promise.<boolean, Error>}
+	 */
+	async update(blocks) {
```

Comment:
this whole thing should be in `submodules/blocks/chain.js`.
instead we need a general way to update the application state

Reply:
This will be to get back to the starting point where system was a chain submodule. Therefore I don't agree.

Reply:
https://github.com/LiskHQ/lisk/issues/3141

Reply:
Yes this should be changed but we can do it as a new issue like you suggest.